### DISCUSSION TITLE
How to transpose MIDI?

### DISCUSSION THREAD

**Discussion by skepsys** [cifkao/html-midi-player#33] [Q&A] [answered]
I'm looking for an ability to adjust the tonic of a melody being loaded into a player (+/- halftone). Any ideas and workarounds on how can I transpose a MIDI file?

**Comment by cifkao** [comment, marked as answer]
All the notes in the file are accessible through `player.noteSequence.notes` (`player.noteSequence` is a `NoteSequence` object from the Magenta.js library). So you can loop through them and change their pitches:
```javascript
for (const note of player.noteSequence.notes) {
  note.pitch += 1;
}
```
Then you will want to assign the note sequence back to the player (and visualizer, if you have one) to make sure it gets updated immediately:
```javascript
player.noteSequence = player.noteSequence;
visualizer.noteSequence = player.noteSequence;
```
